src/content/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-tutorial-advanced-functions.mdx

```diff
@@ -0,0 +1,367 @@
+    src={queriesnrql4tutorial2}
+/>
+    </Step>
+
+With this, you've now learned all aggregation functions that curently exist in NRQL! You can slice and dice your data like a pro. If you think an aggregation ability is missing from NRQL, let your account team know: we're always looking for the next killer functionality that you might need.
+</Steps>
+
+### Use advanced math functions
+
+NRQL also supports some nifty, more advanced mathetmatical functions for those who need to get really deep into their data. These functions can extrapolate values to power of X, find square roots, apply clamping to impose upper and lower bounds, or even just keep values postive on the Y axis.
+
+<Steps>
+    <Step>
+NRQL has many math functions that manipulate values in some way. In this example, you'll see a few demonstrated simultaneously.
+
+The `abs(n)` function returns the absolute value for n: for non-negative n values it returns "n", and for negative n values it returns the positive number "n". For example, `abs(2)` = 2, and `abs(-4)` = 4.
+
+You can also round decimal numbers to integers using `floor()`, `ceil()`, and `round()`. `floor()` returns the closest full integer rounding down, and `ceil()` does the opposite by rounding up. `round()` is bi-directional, and rounds up or down to the closest full integer.
+
+```sql
+SELECT abs(duration), round(duration), ceil(duration), floor(duration) 
+FROM Transaction 
+SINCE 1 day ago
+```
+<img
+    title="Using abs(n)"
+    alt="A screenshot showing the results of a query using absolute value"
+    src={queriesnrql4tutorial3}
+/>
+    </Step>
+
+    <Step>
+
+You can use clamping on an attribute to impose an upper or lower limit on its value. This has use for things like ensuring extreme outliers don't skew the scale of a timeseries graph. `clamp_max(duration, 10)` returns the duration, unless it exceeds 10, in which case it returns 10. Quite simply, anything greater than 10 still returns as to equal 10. `clamp_min(duration,1)` does the inverse; if any duration is below 1, it reports as equal to 1. 
+
+```sql
+SELECT clamp_max(average(duration), 10), clamp_min(average(duration), 1) 
+FROM Transaction 
+SINCE 1 day ago 
+TIMESERIES
+```
+<img
+    title="Clamping data"
+    alt="A screenshot showing a query that uses clamp max"
+    src={queriesnrql4tutorial4}
+/>
+    </Step>
+    <Step>
+Now you can move on to the advanced maths that many find challenging to use. As before, we can show you all of these functions in a single example query.
+
+The `pow()` function raises the first argument to the power of the second argument. In this example, you raise duration to the power of 2. If you need the square root of a value, the `sqrt()` function can quickly give this to you. And the `exp()` function computes the natural exponential function of its argument.
+
+Finally, NRQL also offers logarithms catered to a number of similar functions:
+
+* `ln(n)` computes the natural logarithm: the logarithm base e.
```

Comment:
not a comment but i stared at this for 12 seconds bc it didn't look like a word for a minute